### PR TITLE
Add expected cache-miss criteria

### PR DESCRIPTION
**Pull request by blindpirate**
### Context

We observed some `CACHE_MISS`es and this PR fixes them (hopefully :-)).

### CODE REVIEW

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt [hunk 1]
```diff
@@ -119,7 +119,16 @@ open class BuildScanPlugin : Plugin<Project> {
...
-        isInBuild("Gradle_Check_CompileAll", "Enterprise_Master_Components_GradleBuildScansPlugin_Performance_PerformanceLinux", "Gradle_Check_Gradleception")
+    // 4. buildScanPerformance test, which doesn't depend on compileAll
+    // 5. Gradle/Promotion/Release - Release Nightly Snapshot
+    // 6. Gradle/Promotion/Master - Nightly Snapshot
+        isInBuild("Gradle_Check_CompileAll",
+            "Enterprise_Master_Components_GradleBuildScansPlugin_Performance_PerformanceLinux",
+            "Enterprise_Release_Components_BuildScansPlugin_Performance_PerformanceLinux",
+            "Gradle_Check_Gradleception",
+            "bt61",
+            "bt39"
```
Comment (blindpirate) [inline buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt]: Considering https://github.com/gradle/gradle-private/issues/1795 we can ignore promotion build CACHE_MISSes for now then re-enable them after fixing https://github.com/gradle/gradle-private/issues/1795

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt [hunk 2]
```diff
@@ -119,7 +119,16 @@ open class BuildScanPlugin : Plugin<Project> {
     // 1. CompileAll is the seed build
     // 2. Gradleception which re-builds Gradle with a new Gradle version
     // 3. buildScanPerformance test, which doesn't depend on compileAll
-        isInBuild("Gradle_Check_CompileAll", "Enterprise_Master_Components_GradleBuildScansPlugin_Performance_PerformanceLinux", "Gradle_Check_Gradleception")
+    // 4. buildScanPerformance test, which doesn't depend on compileAll
+    // 5. Gradle/Promotion/Release - Release Nightly Snapshot
+    // 6. Gradle/Promotion/Master - Nightly Snapshot
+        isInBuild("Gradle_Check_CompileAll",
```
Comment (wolfs) [inline buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt]: Nit: I would move `"Gradle_Check_CompileAll"` to a new line.

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt (hunk 1 above)
Comment (wolfs) [reply buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt]: Sounds good.

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt (hunk 2 above)
Comment (blindpirate) [reply buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt]: Done.

**Comment by blindpirate** [comment]
Thanks @wolfs !

**Pull request merged**